index.js

```diff
@@ -99,6 +99,13 @@
          * linked to this component's state.
          */
         componentDidMount: function() {
+          // If we are in an environment without a DOM such 
+          // as shallow rendering or snapshots then we exit 
+          // early to prevent any unhandled errors being thrown.
+          if (!ReactDOM.findDOMNode){
```

Comment:
curious if this is the most authoritative way to determine we have no DOM at all (I would imagine that if ReactDOM is required in correctly, then the findDomNode function at least exists, even if it won't actually work).